src/include/btree_inline.h

```diff
@@ -741,6 +742,13 @@ __wt_tree_modify_set(WT_SESSION_IMPL *session)
 
         S2BT(session)->modified = true;
         WT_FULL_BARRIER();
+        /*
+         *  Simulate a delay after marking a tree dirty - there is a race where checkpoint
+         *  can clear the dirty state before a page is marked clean.
+         */
+        tsp.tv_sec = 0;
+        tsp.tv_nsec = 1000000;
+        __wt_timing_stress(session, WT_TIMING_STRESS_CHECKPOINT_RACE, &tsp);
```

Comment:
I know I asked for this to be added, but now that I see it in the code I've become concerned about the performance consequence of always doing this.

Could you try running a patch build with a call to `WT_DIAGNOSTIC_YIELD` instead of the `timing_stress`? The reason I prefer that is that it doesn't add any overhead in a release build.

To be clear - I'm not sure which way to go, but having information about whether the failure reproduces with that lighter weight mechanism is helpful in deciding.